src/theory/quantifiers/term_database.cpp

```diff
@@ -1034,7 +1034,9 @@ bool TermDb::reset( Theory::Effort effort ){
           eq = itpe->second;
         }
         Trace("quant-ho") << "- assert purify equality : " << eq << std::endl;
-        ee->assertEquality(eq, true, eq);
+        // Note that ee may be the central equality engine, in which case this
+        // equality is explained trivially with "true".
```

Comment:
Add that the explanation does not matter since these are internal terms, which is why doing this is fine (IIRC).

Reply:
Actually, for central equality engine, asserting this equality may trigger a real conflict. "true" is right explanation, since the equality is essentially like saying `purifySkolemFor(t) = t`.